crates/status/src/status_manager.rs

```diff
@@ -16,93 +16,121 @@ pub enum StatusError {
-    csm: watch::Receiver<CsmStatus>,
-    cl: watch::Receiver<ClientState>,
-    l1: watch::Receiver<L1Status>,
-    chs: watch::Receiver<Option<Chainstate>>,
+/// Consolidation of various watch channels.
+pub struct StatusHub {
+    receiver: Arc<StatusReceiver>,
+    sender: Arc<StatusSender>,
 }
 
-impl StatusRx {
-    pub fn new(
-        csm: watch::Receiver<CsmStatus>,
-        cl: watch::Receiver<ClientState>,
-        l1: watch::Receiver<L1Status>,
-        chs: watch::Receiver<Option<Chainstate>>,
-    ) -> Self {
-        Self { csm, cl, l1, chs }
-    }
-
-    pub fn csm(&self) -> &watch::Receiver<CsmStatus> {
-        &self.csm
-    }
-
-    pub fn cl(&self) -> &watch::Receiver<ClientState> {
-        &self.cl
+impl StatusHub {
+    pub fn receiver(&self) -> Arc<StatusReceiver> {
+        self.receiver.clone()
     }
 
-    pub fn l1(&self) -> &watch::Receiver<L1Status> {
-        &self.l1
-    }
-
-    pub fn chs(&self) -> &watch::Receiver<Option<Chainstate>> {
-        &self.chs
+    pub fn sender(&self) -> Arc<StatusSender> {
+        self.sender.clone()
     }
 }
 
-impl StatusTx {
+impl StatusSender {
+    /// Sends the updated `Chainstate` to the chain state receiver. Logs a warning if the receiver
+    /// is dropped.
     pub fn update_chain_state(&self, post_state: &Chainstate) {
         if self.chs.send(Some(post_state.clone())).is_err() {
             warn!("chain state receiver dropped");
         }
     }
 
+    /// Sends the updated `ClientState` to the client state receiver. Logs a warning if the receiver
+    /// is dropped.
     pub fn update_client_state(&self, post_state: &ClientState) {
         if self.cl.send(post_state.clone()).is_err() {
             warn!("client state receiver dropped");
         }
     }
 
+    /// Sends the updated `L1Status` to the L1 status receiver. Logs a warning if the receiver is
+    /// dropped.
     pub fn update_l1_status(&self, post_state: &L1Status) {
         if self.l1.send(post_state.clone()).is_err() {
             warn!("l1 status receiver dropped");
         }
     }
 
-    pub fn update_csm_status(&self, post_state: &CsmStatus) {
-        if self.csm.send(post_state.clone()).is_err() {
-            warn!("csm status receiver dropped");
-        }
+    /// Subscribes to updates for `ClientState` and returns a receiver.
+    pub fn subscribe_cl(&self) -> watch::Receiver<ClientState> {
+        self.cl.subscribe()
+    }
+}
+
+impl StatusReceiver {
+    /// Borrow [`ClientState`] from channel. Be careful while using this to avoid deadlock which can
+    /// occur if the borrow lives too long during which sender tries to send through this.
+    pub fn borrow_client_state(&self) -> Ref<ClientState> {
+        self.cl.borrow()
+    }
+
+    /// Gets the lastest [`ClientState`].
+    pub fn client_state(&self) -> ClientState {
+        self.cl.borrow().clone()
+    }
+
+    /// Borrow [`Chainstate`] from channel. Be careful while using this to avoid deadlock which can
```

Comment:
```suggestion
    /// Borrows [`Chainstate`] from channel. Be careful while using this to avoid deadlock which can
```

You are using "Gets" above, so be consistent.